python/lsst/daf/butler/registry/queries/expressions/parser/parserYacc.py

```diff
@@ -108,6 +108,11 @@ def _parseTimeString(time_str):
     ValueError
         Raised if input string has unexpected format
     """
+    # Check for time zone. Python datetime objects can be timezone-aware
+    # and if one has been stringified then there will be a +00:00 on the end.
+    # Special case UTC. Fail for other timezones.
+    time_str = time_str.replace("+00:00", "")
```

Comment:
I'm pretty sure the `+/-\d\d:\d\d` is not locale-specific for timezones in ISO standard.

This code will fail if a non-UTC timezone turns up because Astropy will not like it. I think the error message is a bit opaque though because it hides the complaint from astropy itself. Maybe it is better to change the regex to optional `+00:00` and strip it here since that will trigger the error message higher up.

Reply:
I saw your recent commit - looks like you are still stripping `+00:00` before doing regex check, and there is no reason for regex to have timezone in it if you strip it beforehand?

Reply:
I thought that stripping happened after the regex check... I am going to check again.

Reply:
Ha. Okay. Now I see. The regex triggers after the stripping in that astropy routine. I will fix to only do the stripping.